guides/release/components-update/component-state.md

```diff
@@ -0,0 +1,306 @@
+<p>{{this.count}}</p>
+<p>× {{this.multiple}}</p>
+<p>= {{this.total}}</p>
+
+<button {{on "click" (fn this.change 1)}}>+1</button>
+<button {{on "click" (fn this.change -1)}}>-1</button>
+
+<button {{on "click" this.double}}>Double It</button>
+```
+
+And we're all done. Try to click the plus, minus and double buttons in any order, and watch as the three outputs stay up to date perfectly.
+
+<div class="cta">
+  <div class="cta-note">
+    <div class="cta-note-body">
+      <div class="cta-note-heading">Zoey says...</div>
+      <div class="cta-note-message">
+        <p>
+          You might have been tempted to make <code>total</code> a <code>@tracked</code> property and update it in
+          the <code>double</code> and <code>change</code> actions. But this kind of "push-based" approach creates a
+          lot of bugs. What happens if you create a new way to update <code>multiple</code> or <code>amount</code>
+          properties and forget to update <code>total</code> at the same time?
+        </p>
+        <p>
+          When you use getters and functions to <em>derive</em> the state you need, you're taking advantage of
+          the benefits of <strong>declarative</strong> programming. In declarative programming, you describe
+          <em>what</em> you need, not <em>how</em> to get it, which reduces the number of places where you can
+          make mistakes.
+        </p>
+        <p>Making a <code>total</code> getter that computed the total from the <code>amount</code> and
+        <code>multiple</code> properties was more <strong>declarative</strong> than setting <code>total</code>
+        in all of the places that could have affected it (changing <code>total</code> directly is called the
+        "imperative" approach).</p>
+      </div>
+    </div>
+    <img src="/images/mascots/zoey.png" role="presentation" alt="">
+  </div>
+</div>
+
+## Combining Arguments and State
+
+Instead of allowing the component itself to be responsible for the multiple, let's allow it to be passed in.
+
+```handlebars {data-filename="app/components/double-it.hbs"}
+<Counter @multiple={{this.multiple}} />
+
+<button {{on "click" this.double}}>Double It</button>
+```
+
+```js {data-filename="app/components/double-it.js"}
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+
+export default class extends Component {
+  @tracked multiple = 1;
+
+  @action
+  change(amount) {
+    this.multiple = this.multiple * 2;
+  }
+}
+```
+
+In the `Counter` component, instead of tracking the `multiple` internally, we take it as an argument. In the template, we refer to the argument as `@multiple`.
+
+```handlebars {data-filename="app/components/counter.hbs"}
+<p>{{this.count}}</p>
+<p>× {{@multiple}}</p>
+<p>= {{this.total}}</p>
+
+<button {{on "click" (fn this.change 1)}}>+1</button>
+<button {{on "click" (fn this.change -1)}}>-1</button>
+```
+
+In JavaScript, we refer to it as `this.args.multiple`.
```

Comment:
We need to explain `this.args` here. We've never seen this before, so it feels randomly dropped in here.